[ci] add tests for examples

erkue opened the pull request:
Adds a test for every example in the docs using `tytanic`. They are contained in `tests/examples`.

Additionally, adds a workflow that can be manually dispatched to update every test.

Mc-Zen commented on .github/workflows/run_tests.yml:
```diff
@@ -22,17 +25,27 @@ jobs:
-          git: https://github.com/tingerrr/typst-test.git
-          tag: ci-semi-stable
-
       - name: Setup typst
```
The setup typst step is now not needed since tytanic brings its own version of typst

Mc-Zen commented on .github/workflows/update_test_refs.yml:
```diff
@@ -0,0 +1,34 @@
+      - name: Setup typst
+        uses: typst-community/setup-typst@v3
+        with:
+          typst-version: 'v0.12.0'
```
same here

Mc-Zen commented on tests/examples/dual_axis/test.typ:
```diff
@@ -0,0 +1,33 @@
+#set page(width: auto, height: auto, margin: 0pt)
+#import "/src/lilaq.typ" as lq
+
+//#let (λ, intensity) = lq.load-txt(read("spectrum_o2.txt"), skip-rows: 2, delimiter: " ")
```
ah, yes the corresponding file can be found at https://github.com/lilaq-project/lilaq-project.github.io/blob/main/docs/examples/spectrum_o2.txt

I think we can just copy it into the test folder and use the proper data, so that the energy axis doesn't get messed up.

Mc-Zen commented:
Thank you, this is truly awesome! 🎉

I wonder whether we should add a small page margin (instead of keeping it to `0pt`) but actually it's not really necessary.

erkue commented:
I removed the unnecessary Typst installation and added a small margin to the test images for improved layout.

Also, I have realized that the climograph test image does not look right. Do you have an idea @Mc-Zen?

Mc-Zen commented:
We're already earning fruits from your PR, it seems. 

 I bisected the issue and it's due to dc09c25f94a7cb2b114b88421a1923981372e7a4.

Mc-Zen commented:
Actually, I found the issue, it was simply a typo. Please rebase and update the reference images again!

erkue commented:
Looks good now—no issues on my end.

Mc-Zen commented:
Thank you 🎉
